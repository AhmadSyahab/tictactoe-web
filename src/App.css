.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}
.table-primary,
.table-primary>th,
.table-primary>td {
  background-color: rgba(255, 255, 255, 0.2) !important;
}

.table.vr-table,
.table.vr-table>th,
.table.vr-table>td {
  background-color: rgba(255, 255, 255, 0.32) !important;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
