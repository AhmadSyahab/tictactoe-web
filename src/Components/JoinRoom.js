import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import { db } from '../firebase.js'

export default class JoinRoom extends Component {
  constructor(props) {
    super()
    this.state = {
      uid: localStorage.getItem('uid'),
      player: [],
      gameKey: '',
      gameKeyDummy: '',
      type: '',
      games: [],
    }
  }

  handleButton (e) {
    console.log(e);
  }

  joinRoom(gameRoom) {
    db.ref('games').child(gameRoom).once('value', room => {
      if(room.val().player2.uid === '' && room.val().winner === '' &&
         room.val().player1.uid !== localStorage.getItem('uid') &&
         room.val().player2.uid !== localStorage.getItem('uid')) {
        db.ref('games').child(gameRoom).update({
          player2: {
            name: this.state.player.name,
            uid: this.state.player.uid,
            type: 'O',
            avatar: this.state.player.avatar
          }
        })

        window.location.href = `http://localhost:8081/vr/?room=${gameRoom}&player=${this.state.uid}`;
      }else if(room.val().winner !== ''){
        alert('game already finished')
      }else if(room.val().player2.uid !== '' && room.val().winner === ''){
        alert('game already Full')
      }else if(room.val().player1.uid === localStorage.getItem('uid')){
        alert('Welcome Back')
        window.location.href = `http://localhost:8081/vr/?room=${gameRoom}&player=${this.state.uid}`;
      }
    })
  }

  componentDidMount() {
    db.ref('users').orderByChild('uid').equalTo(this.state.uid).once('value', snaphotUser => {
      snaphotUser.forEach(snapUser => {
        this.state.player = snapUser.val()
      })
    })

    db.ref('games').on('value', snapshot => {
      let allGames = []
      for(let game in snapshot.val()){
        let objGame= {
          gameName: snapshot.val()[game].roomName,
          gameId: game
        }
        allGames.push(objGame)
      }
      this.setState({
        games: allGames
      })
    })
  }

  render () {
    return (
      <div className='col-md-8 offset-md-2'>
        <h1>Select Room</h1>
        <div className="list-room" data-toggle="buttons">
          { 
            this.state.games.map(game => {
              return <button type="button" className="join-list-btn v-button" onClick={()=> this.joinRoom(game.gameId)}>{game.gameName}</button>
            })
          }
        </div>
        <Link to="/play"><button className='back-button'></button></Link>
      </div>
    )
  }
}