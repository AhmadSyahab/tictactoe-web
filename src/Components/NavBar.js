import React, { Component } from 'react';
import {
  Link
} from 'react-router-dom';

export default class NavBar extends Component {
  constructor (props) {
    super (props)
    this.state ={
      statusLogin: false
    }
  }

  loginHandle () {
    this.setState({statusLogin: !this.state.statusLogin})
    console.log(this.state.statusLogin);
  }

  render () {
    const statusLogin = this.state.statusLogin
    return (
      <nav className="navbar navbar-expand-lg navbar-dark bg-primary">
        <Link className="navbar-brand" to="/">Navbar</Link>
        <button className="navbar-toggler" type="button" data-toggle="collapse"
         data-target="#navbarColor01" aria-controls="navbarColor01" 
         aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon"></span>
        </button>

        <div className="collapse navbar-collapse" id="navbarColor01">
          <ul className="navbar-nav mr-auto">
            <li className="nav-item active">
              <Link className="nav-link" to="/">Home <span className="sr-only">(current)</span></Link>
            </li>
          </ul>
          <form className="form-inline my-2 my-lg-0">
            {!statusLogin ? <button className="btn btn-secondary my-2 my-sm-0" type="submit">Login</button>
              : <button className="btn btn-secondary my-2 my-sm-0" type="submit">Logout</button>  
          }   
          </form>
        </div>
      </nav>
    )
  }
}