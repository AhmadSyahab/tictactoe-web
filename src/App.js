import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route
} from 'react-router-dom';
import {Provider} from 'react-redux';

import './App.css';

import store from './store'

import PlayGame from './Components/PlayGame'
import LoginForm from './Components/LoginForm'
import CreateRoom from './Components/CreateRoom'
import JoinRoom from './Components/JoinRoom';
import SelectAvatar from './Components/SelectAvatar';
import NavBar from './Components/NavBar';
import Footer from './Components/Footer';
import History from './Components/History';

class App extends Component {
  render() {
    return (
      <Provider store={store} >
        <Router>
          <div className="App">
            <Route exact path ="/" component={LoginForm} />
            <Route path ="/avatar" render={() => <NavBar><SelectAvatar/></NavBar>} />
            <Route exact path ="/play" render={() => <NavBar><PlayGame/></NavBar>} />
            <Route path ="/play/create" render={() => <NavBar><CreateRoom/></NavBar>} />
            <Route path ="/play/join" render={() => <NavBar><JoinRoom/></NavBar>} />
            <Route path ="/history" render={() => <NavBar><History /></NavBar>} />
            <Footer/> 
          </div>
        </Router>
      </Provider>
    );
  }
}

export default App;
