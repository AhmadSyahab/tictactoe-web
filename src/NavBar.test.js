import React from 'react';
import ReactDOM from 'react-dom';
import NavBar from './Components/NavBar';
import JoinRoom from './Components/JoinRoom';
import {Route} from 'react-router-dom'
import LoginForm from './Components/LoginForm'
import App from './App';
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe('Testing the NavBar', () => {
  test('It should render the button', () => {
    const Link = shallow(
      <button className="navbar-toggler" type="button" data-toggle="collapse"
        data-target="#navbarColor01" aria-controls="navbarColor01"
        aria-expanded="false" aria-label="Toggle navigation">
        <span className="navbar-toggler-icon"></span></button>
    )
    expect(Link).toMatchSnapshot()
  })
  test('It should render the form', () => {
    const Link = shallow(
      <form className="form-inline my-2 my-lg-0">
        {<button className="btn btn-danger my-2 my-sm-0" type="submit" >Logout</button>}
      </form>
    )
    expect(Link).toMatchSnapshot()
  })
  test('Should render nav', () => {
    const wrapper = shallow(<div />)
    const nav = wrapper.find('nav')
    expect(nav).toBeDefined()
  })
  test('It Should have Login Form', () => {
    const app = shallow(<App />)
    const loginForm = app.find(<Route exact path="/" component={LoginForm} />)
    expect(loginForm).toMatchSnapshot()
  })
  test('App should have PlayGame', () => {
    const app = shallow(<App />)
    const playGame = app.find(<Route exact path="/play" render={() => <NavBar><PlayGame /></NavBar>} />)
    expect(playGame).toMatchSnapshot()
  })
})