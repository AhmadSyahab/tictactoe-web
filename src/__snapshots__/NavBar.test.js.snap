// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the NavBar It should render the button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <button
    aria-controls="navbarColor01"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarColor01"
    data-toggle="collapse"
    type="button"
>
    <span
        className="navbar-toggler-icon"
    />
</button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-controls": "navbarColor01",
      "aria-expanded": "false",
      "aria-label": "Toggle navigation",
      "children": <span
        className="navbar-toggler-icon"
/>,
      "className": "navbar-toggler",
      "data-target": "#navbarColor01",
      "data-toggle": "collapse",
      "type": "button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "navbar-toggler-icon",
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-controls": "navbarColor01",
        "aria-expanded": "false",
        "aria-label": "Toggle navigation",
        "children": <span
          className="navbar-toggler-icon"
/>,
        "className": "navbar-toggler",
        "data-target": "#navbarColor01",
        "data-toggle": "collapse",
        "type": "button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "navbar-toggler-icon",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Testing the NavBar It should render the form 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <form
    className="form-inline my-2 my-lg-0"
>
    <button
        className="btn btn-danger my-2 my-sm-0"
        type="submit"
    >
        Logout
    </button>
</form>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="btn btn-danger my-2 my-sm-0"
        type="submit"
>
        Logout
</button>,
      "className": "form-inline my-2 my-lg-0",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Logout",
        "className": "btn btn-danger my-2 my-sm-0",
        "type": "submit",
      },
      "ref": null,
      "rendered": "Logout",
      "type": "button",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="btn btn-danger my-2 my-sm-0"
          type="submit"
>
          Logout
</button>,
        "className": "form-inline my-2 my-lg-0",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Logout",
          "className": "btn btn-danger my-2 my-sm-0",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Logout",
        "type": "button",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
